test/framework/clusterctl/client.go

```diff
@@ -0,0 +1,155 @@
+	defer log.Close()
+
+	_, err := clusterctlClient.Init(initOpt)
+	Expect(err).ToNot(HaveOccurred(), "failed to run clusterctl init")
+}
+
+// ConfigClusterInput is the input for ConfigCluster.
+type ConfigClusterInput struct {
+	LogPath                  string
+	ClusterctlConfigPath     string
+	KubeconfigPath           string
+	InfrastructureProvider   string
+	ClusterName              string
+	KubernetesVersion        string
+	ControlPlaneMachineCount *int64
+	WorkerMachineCount       *int64
+}
+
+// ConfigCluster gets a workload cluster based on a template.
+func ConfigCluster(ctx context.Context, input ConfigClusterInput) []byte {
+	By(fmt.Sprintf("clusterctl config cluster %s --infrastructure %s --kubernetes-version %s --control-plane-machine-count %d --worker-machine-count %s",
+		input.ClusterName,
+		input.InfrastructureProvider,
+		input.KubernetesVersion,
+		input.ControlPlaneMachineCount,
+		input.WorkerMachineCount,
+	))
+
+	templateOptions := clusterctlclient.GetClusterTemplateOptions{
+		Kubeconfig: input.KubeconfigPath,
+		ProviderRepositorySource: &clusterctlclient.ProviderRepositorySourceOptions{
+			InfrastructureProvider: input.InfrastructureProvider,
+			Flavor:                 "",
+		},
+		ClusterName:              input.ClusterName,
+		KubernetesVersion:        input.KubernetesVersion,
+		ControlPlaneMachineCount: input.ControlPlaneMachineCount,
+		WorkerMachineCount:       input.WorkerMachineCount,
+	}
+
+	clusterctlClient, log := getClusterctlClientWithLogger(input.ClusterctlConfigPath, "clusterctl-config-cluster.log", input.LogPath)
+	defer log.Close()
+
+	template, err := clusterctlClient.GetClusterTemplate(templateOptions)
+	Expect(err).ToNot(HaveOccurred(), "Failed to run clusterctl config cluster")
+
+	yaml, err := template.Yaml()
+	Expect(err).ToNot(HaveOccurred(), "Failed to generate yaml for the workload cluster template")
+
+	log.WriteString(string(yaml))
+
+	return yaml
+}
+
+// MoveInput is the input for ClusterctlMove.
+type MoveInput struct {
+	LogPath              string
+	ClusterctlConfigPath string
+	FromKubeconfigPath   string
+	ToKubeconfigPath     string
+	Namespace            string
+}
+
+// ClusterctlMove moves a workloads clusters.
+func ClusterctlMove(ctx context.Context, input MoveInput) {
+	By("Moving workload clusters")
+
+	clusterctlClient, log := getClusterctlClientWithLogger(input.ClusterctlConfigPath, "clusterctl-move.log", input.LogPath)
+	defer log.Close()
+
+	options := clusterctlclient.MoveOptions{
+		FromKubeconfig: input.FromKubeconfigPath,
+		ToKubeconfig:   input.ToKubeconfigPath,
+		Namespace:      input.Namespace,
+	}
+
+	Expect(clusterctlClient.Move(options)).To(Succeed(), "Failed to run clusterctl move")
+}
+
+func getClusterctlClientWithLogger(configPath, logName, logPath string) (clusterctlclient.Client, *logger.LogFile) {
+	log := logger.CreateLogFile(logger.CreateLogFileInput{
+		LogPath: logPath,
+		Name:    logName,
```

Comment:
Do we want to keep clusterctl logs from different e2e runs like we do in docker e2e (it differentiates log file names by using container names)?
If so we may add some randomized addition at the end of logName.
This can be addressed in a follow-up enhancement PR.

Reply:
My assumption is that each test will produce more files than now (clusterctl logs, object dump & controller logs), and so files should be grouped in folders.
If this is not enough, we can consider making the file name configurable as well in a follow-up PR (random additions don't help you to understand e.g. the order of actions)